pkg/apiserver/bootstrap.go

```diff
@@ -0,0 +1,142 @@
+	"context"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/replicated-sdk/pkg/appstate"
+	appstatetypes "github.com/replicatedhq/replicated-sdk/pkg/appstate/types"
+	"github.com/replicatedhq/replicated-sdk/pkg/heartbeat"
+	"github.com/replicatedhq/replicated-sdk/pkg/helm"
+	"github.com/replicatedhq/replicated-sdk/pkg/k8sutil"
+	sdklicense "github.com/replicatedhq/replicated-sdk/pkg/license"
+	"github.com/replicatedhq/replicated-sdk/pkg/mock"
+	"github.com/replicatedhq/replicated-sdk/pkg/store"
+	"github.com/replicatedhq/replicated-sdk/pkg/upstream"
+	upstreamtypes "github.com/replicatedhq/replicated-sdk/pkg/upstream/types"
+	"github.com/replicatedhq/replicated-sdk/pkg/util"
+)
+
+func bootstrap(params APIServerParams) error {
+	verifiedLicense, err := sdklicense.VerifySignature(params.License)
+	if err != nil {
+		return backoff.Permanent(errors.Wrap(err, "failed to verify license signature"))
+	}
+
+	if !util.IsAirgap() {
+		// sync license
+		licenseData, err := sdklicense.GetLatestLicense(verifiedLicense, params.ReplicatedAppEndpoint)
+		if err != nil {
+			return errors.Wrap(err, "failed to get latest license")
+		}
+		verifiedLicense = licenseData.License
+	}
+
+	// check license expiration
+	expired, err := sdklicense.LicenseIsExpired(verifiedLicense)
+	if err != nil {
+		return errors.Wrap(err, "failed to check if license is expired")
+	}
+	if expired {
+		return backoff.Permanent(errors.New("License is expired"))
+	}
+
+	// generate / retrieve sdk and app ids
+	replicatedID, appID, err := util.GenerateIDs(params.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to generate ids")
+	}
+
+	channelID := params.ChannelID
+	if channelID == "" {
+		channelID = verifiedLicense.Spec.ChannelID
+	}
+
+	channelName := params.ChannelName
+	if channelName == "" {
+		channelName = verifiedLicense.Spec.ChannelName
+	}
+
+	storeOptions := store.InitInMemoryStoreOptions{
+		ReplicatedID:          replicatedID,
+		AppID:                 appID,
+		License:               verifiedLicense,
+		LicenseFields:         params.LicenseFields,
+		AppName:               params.AppName,
+		ChannelID:             channelID,
+		ChannelName:           channelName,
+		ChannelSequence:       params.ChannelSequence,
+		ReleaseSequence:       params.ReleaseSequence,
+		ReleaseCreatedAt:      params.ReleaseCreatedAt,
+		ReleaseNotes:          params.ReleaseNotes,
+		VersionLabel:          params.VersionLabel,
+		ReplicatedAppEndpoint: params.ReplicatedAppEndpoint,
+		Namespace:             params.Namespace,
+	}
+	if err := store.InitInMemory(storeOptions); err != nil {
+		return errors.Wrap(err, "failed to init store")
+	}
+
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		return errors.Wrap(err, "failed to get clientset")
+	}
+
+	if store.GetStore().IsDevLicense() {
+		mock.InitMock(clientset, store.GetStore().GetNamespace())
+	}
+
+	isMockEnabled, err := mock.MustGetMock().IsMockEnabled(context.TODO())
```

Comment:
can we create a real context to use here (and potentially elsewhere we use TODO)?

Reply:
we could...

Reply:
doesn't have to be in this PR, but likely something we should do at some point

Reply:
done